spec/system/admin/reports/sales_tax/sales_tax_totals_by_producer_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+describe "Sales Tax Totals By Producer" do
+  #  Scenarion 1:
+  #  1 producer
+  #  1 distributor
+  #  1 product that costs 100$
+  #  1 order with 1 line item
+  #  the line item match 2 tax rates: country (2.5%) and state (1.5%)
+  let!(:state_zone){ create(:zone_with_state_member) }
+  let!(:country_zone){ create(:zone_with_member) }
+  let!(:tax_category){ create(:tax_category) }
+  let!(:state_tax_rate){ create(:tax_rate, zone: state_zone, tax_category: tax_category) }
+  let!(:country_tax_rate){ create(:tax_rate, zone: country_zone, tax_category: tax_category) }
+  let!(:ship_address){ create(:ship_address) }
+
+  let!(:variant){ create(:variant) }
+  let!(:product){ variant.product }
+  let!(:supplier){ create(:supplier_enterprise) }
+  let!(:distributor){ create(:distributor_enterprise_with_tax) }
+  let!(:payment_method){ create(:payment_method, :flat_rate) }
+  let!(:shipping_method){ create(:shipping_method, :flat_rate) }
+
+  let!(:order){ create(:order_with_distributor, distributor: distributor) }
```

Comment:
I think it could be interesting to test some filters.

We could, for example, create two orders for two different customers within the same order cycle, and assure that `order1` displays the data correctly for `customer1`, and `order2` for `customer2`, within the same order cycle.

Customer is also a bit of a sensitive information. I'm thinking that `supplier1` should not see the email addresses of `supplier2`, within the same order cycle... In this test, we're testing as superadmin, which should see all, but I think we might need to assure that customer data is not shared between supplier profiles; OC coordinator (shop) should be able to see all.

Reply:
What do you think @abdellani @lin-d-hop ?

Reply:
@filipefurtad0 

> We could, for example, create two orders for two different customers within the same order cycle, and assure that order1 displays the data correctly for customer1, and order2 for customer2, within the same order cycle.

I like the idea of testing the filters, I'll implement it in a different `RSpec.context`.

>  I'm thinking that supplier1 should not see the email addresses of supplier2, within the same order cycle... 

I agree, but are you saying that we should remove this filter if the user is not a distributor or a super admin?

Reply:
I implemented the following scenarios

oc1  has 2 orders:
order 1 is done by customer 1
order 2 is done by customer 2

I generated the report with the following parameters:
1. no filters
2. filtered customer 1
3. filtered customer 2
4. filter customer 1 & customer 2

Reply:
> but are you saying that we should remove this filter if the user is not a distributor or a super admin?

No, my suggestion was just regarding the test/spec, and not the feature itself. In anycase, I think we would need that filter, in case different customers placed orders for OCs supplied by a given supplier.